### DISCUSSION TITLE
Feature Suggestion: Search and Replace Rules for Pasting

### DISCUSSION THREAD

**Discussion by Makki93** [p0deje/Maccy#855] [Ideas]
Hi,

I would like to suggest a new feature for Maccy.

As a programmer, I often copy similarly formatted strings or other content. When pasting this content, I frequently need to adapt it slightly. A common example is switching a network file path format from Windows to macOS.

It would be incredibly helpful if Maccy could include a feature that allows users to create regex or normal search and replace rules. These rules could be toggled on individually and applied via a shortcut that pastes the content after internally running those rules.

What do you think about this idea?

Thank you!

**Comment by p0deje** [comment]
It falls into the category of Shortcuts.app workflows from my point of view. There is a way to get the contents of the selected copied item which means you can then manipulate it as much as you want.

For example, this is how Trim and Paste is implemented. You can do something similar for your case.
<img width="1312" alt="Screenshot 2024-08-29 at 08 04 13" src="https://github.com/user-attachments/assets/0901ff51-a556-4892-b172-e5ab87b252a8">
